### ISSUE TITLE
CPP Library

### ISSUE THREAD

**Issue by purgeme** [Mario-td/Simplified-hand-tracking-with-Mediapipe-CPP#2]
Hey, so I have tried shared memory for the same purpose before but had some issues and moved to creating a dynamic library and using that to run the graphs.

Code: https://github.com/purgeme/mediapipe_cpp_lib

My primary source of insight has been this: https://github.com/wongfei/ue4-mediapipe-plugin with https://github.com/wongfei/mediapipe/tree/unreal

I am not very familiar with how APIs should be designed and what functionality people require, could you help me with that ?

**Comment by Mario-td** [comment]
@purgeme what kind of issues did you have? I tried to make a shared library to compile it later with CMake but I didn't succeed. What sort of API do you mean?

**Comment by purgeme** [comment]
Since it is running as a different program, so running and closing it. Just adds an extra step of complexity.
Then I wasn't very familiar with the memory handling in cpp.
My way of using shared memory was allocating a block of it, combining landmarks into a single string, passing that string, then splitting it. This was too expensive.
Plus why I did it like this was that due to lack of mediapipe documentation for cpp, I was running mediapipe in python and then sharing data to a cpp program ( Needed it in cpp for a godot module ).
EDIT: there was also the problem of synchronizing 2 different programs, synchronizing threads seems easier.

Later, I took time to go through github issues and mediapipe code to learn its working and built a shared library with bazel to use it for the Godot module. Now, since mediapipe doesn't have an API/library for cpp. I am thinking of making it so that my cpp library can be used universally as an unofficial cpp lib for mediapipe. But, again I am not a professional programmer yet and lack understanding of proper code structure, best practices and stuff.
